library/core/src/ops/try_trait.rs

```diff
@@ -460,14 +461,17 @@ pub(crate) struct NeverShortCircuit<T>(pub T);
 impl<T> NeverShortCircuit<T> {
     /// Wrap a binary `FnMut` to return its result wrapped in a `NeverShortCircuit`.
     #[inline]
-    pub fn wrap_mut_2<A, B>(mut f: impl FnMut(A, B) -> T) -> impl FnMut(A, B) -> Self {
-        move |a, b| NeverShortCircuit(f(a, b))
+    pub const fn wrap_mut_2_imp<A, B, F: ~const FnMut(A, B) -> T>(
```

Comment:
Request for the future: please don't rename things like this, since that causes unnecessary conflicts later.

Reply:
Considering this changed the arguments it would cause breakage either way, wouldn't it?

Reply:
And I just noticed, we should probably fix that doc comment.

Reply:
Oh, my bad, I'd missed that it doesn't do what it used to.  Yeah, updating the comment to reflect that probably would have made me notice that.

I take it it can't return `-> impl ~const Fn…`?

Reply:
I fixed the documentation in #102276 since its kind of related.